lib/spack/spack/test/build_systems.py

```diff
@@ -119,12 +125,9 @@ def test_cmake_std_args(config, mock_packages):
 
 
 def test_cmake_bad_generator(config, mock_packages):
-    s = Spec('cmake-client')
-    s.concretize()
-    pkg = spack.repo.get(s)
-    pkg.generator = 'Yellow Sticky Notes'
-    with pytest.raises(spack.package.InstallError):
-        get_std_cmake_args(pkg)
+    with pytest.raises(serr.SpackError):
+        s = Spec('cmake-client generator="Yellow Sticky Note"')
```

Comment:
I think the `generator` variant was removed from `CMakePackage`, and also all of these tests were disabled, so I think this can be reverted for those two reasons.

Reply:
Handled.